src/trans.rs

```diff
@@ -844,13 +1195,75 @@ fn rust_ty_to_binaryen<'tcx>(t: Ty<'tcx>) -> BinaryenType {
 fn sanitize_symbol(s: &str) -> String {
     s.chars().map(|c| {
         match c {
-            '<' | '>' | ' ' => '_',
+            '<' | '>' | ' ' | '(' | ')' => '_',
```

Comment:
This can be in a followup PR, but we probably want this to make different characters to different strings. As it stands now, something like `foo<bar>` and `foo(bar)` both map to `foo_bar_`, which is also a name someone can type. This could potentially lead to weird bugs caused by name collisions.

I suspect using a reasonably common C++ name mangling scheme might be the right approach here.